### PR TITLE
instance config: add custom_auth0_domain parameter

### PR DESCRIPTION
**Pull request by jgehrcke**
The `custom_auth0_domain` parameter logically accompanies `custom_auth0_client_id`.

`custom_auth0_client_id` was not generally functional w/o the domain parameter.

Also see https://github.com/opstrace/opstrace/issues/1175

Tested config validation with an intentionally bad config:

```
± node packages/cli/build/index.js create aws rofl -c cluster-config.yaml 
2021-09-08T14:42:30.744Z info: logging to file: opstrace_cli_create_20210908-144230Z.log
2021-09-08T14:42:30.746Z info: Discovered AWS credentials. Access key: AKIA...AOGY
2021-09-08T14:42:30.763Z error: invalid config document: 'custom_auth0_client_id' requires 'custom_auth0_domain' and vice versa
```

### CODE REVIEW

**Comment by jgehrcke** [comment]
Two screenshots showing the Auth0  "tenant domain" that this parameter is about:

Upon creating an Auth0 tenant:
Screenshot from 2021-09-08 16-50-30

When configuring an 'application' within a tenant:
Screenshot from 2021-09-08 16-51-45

**Comment by jgehrcke** [comment]
Debugging the last `test-upgrades-gcp` failure.


### initial cluster create succeeded

```
[2021-09-09T19:24:14Z] 2021-09-09T19:24:14.966Z [34mdebug[39m: task `cluster creation` succeeded (attempt 1)
[2021-09-09T19:24:14Z] 2021-09-09T19:24:14.966Z [34mdebug[39m: end of createCluster()
[2021-09-09T19:24:14Z] 2021-09-09T19:24:14.966Z [34mdebug[39m: shut down logger, then exit with code 0
```

### upgrade: new cluster config

```
[2021-09-09T19:24:18Z] 2021-09-09T19:24:18.040Z [34mdebug[39m: user-given cluster config parsed. JSON representation:
[2021-09-09T19:24:18Z] {
[2021-09-09T19:24:18Z]   "tenants": [
[2021-09-09T19:24:18Z]     "default"
[2021-09-09T19:24:18Z]   ],
[2021-09-09T19:24:18Z]   "env_label": "opstrace-ci",
[2021-09-09T19:24:18Z]   "cert_issuer": "letsencrypt-staging",
[2021-09-09T19:24:18Z]   "aws": {
[2021-09-09T19:24:18Z]     "eks_admin_roles": [
[2021-09-09T19:24:18Z]       "AWSReservedSSO_AdministratorAccess_8488c3da2f880f06"
[2021-09-09T19:24:18Z]     ]
[2021-09-09T19:24:18Z]   },
[2021-09-09T19:24:18Z]   "custom_auth0_client_id": "5MoCYfPXPuEzceBLRUr6T6SAklT2GDys",
[2021-09-09T19:24:18Z]   "custom_auth0_domain": "opstrace-dev.us.auth0.com"
[2021-09-09T19:24:18Z] }
```

looks about right (with the custom_auth0_domain param).

### infra upgrade worked

```
[2021-09-09T19:25:36Z] 2021-09-09T19:25:36.018Z [34mdebug[39m: task `cluster infrastructure upgrade` succeeded (attempt 1)
```

### controller config upgrade: missing param

```
[2021-09-09T19:25:36Z] 2021-09-09T19:25:36.027Z [34mdebug[39m: upgraded controller config {
[2021-09-09T19:25:36Z]   "name": "prs-bk-5786-3c5-ug",
...
[2021-09-09T19:25:36Z]   "disable_data_api_authentication": false,
[2021-09-09T19:25:36Z]   "custom_auth0_client_id": "5MoCYfPXPuEzceBLRUr6T6SAklT2GDys",
[2021-09-09T19:25:36Z]   "cliMetadata": {
...
```
this is missing the custom_auth0_domain param. Why?

Because as an input/output consideration:

- as part of the upgrade procedure the new controller config seems to be built solely based on the old controller config
- in other words, the user-given Opstrace instance config file is currently ignored when building the new controller config

Two quotes of mine from https://github.com/opstrace/opstrace/issues/326#issuecomment-806844789:

> My idea / hope here was that with the hibernation upgrade method that controller config migration is not needed, because the new controller config can be deterministically derived from the new input (the new cluster config), and maybe some additional heuristics.

> That is: let's explore having the "installer/upgrader" replace the controller config brutally. Maybe this will require changing things elsewhere, but it would be a lovely invariant to have towards simplicity.
